### ISSUE TITLE
Import Error: private method warn called for nil:NilClass

### ISSUE THREAD

**Issue by MattMencel** [zdennis/activerecord-import#492]
I'm trying this in a standalone program, not Rails.  My test DB is SQLite3 where I see this error.  My prod db is a MariaDB database and I'm using the mysql2 gem, but I haven't tested it there yet.  I'm not sure what I'm doing wrong here... is it SQLite3 not supporting the on_key_duplicate_update? It seems ar-import is seeing a nil somewhere...

```
puts columns
]
    [0] :id,
    [1] :birthdate,
    [2] :givenname,
    [3] :middlename,
    [4] :sn,
    [5] :suffix
]

puts values
[
    [0] [
        [0] "12345",
        [1] Fri, 05 Aug 1977,
        [2] "Joe",
        [3] "",
        [4] "Smith",
        [5] ""
    ]
]

Person.import columns, values, on_duplicate_key_update: [ :id, :birthdate, :givenname, :middlename, :sn, :suffix ]

/Users/matt/.chefdk/gem/ruby/2.4.0/gems/activerecord-import-0.22.0/lib/activerecord-import/adapters/abstract_adapter.rb:50:in `post_sql_statements': private method `warn' called for nil:NilClass (NoMethodError)
	from /Users/matt/.chefdk/gem/ruby/2.4.0/gems/activerecord-import-0.22.0/lib/activerecord-import/import.rb:709:in `import_without_validations_or_callbacks'
	from /Users/matt/.chefdk/gem/ruby/2.4.0/gems/activerecord-import-0.22.0/lib/activerecord-import/import.rb:662:in `import_with_validations'
	from /Users/matt/.chefdk/gem/ruby/2.4.0/gems/activerecord-import-0.22.0/lib/activerecord-import/import.rb:602:in `import_helper'
	from /Users/matt/.chefdk/gem/ruby/2.4.0/gems/activerecord-import-0.22.0/lib/activerecord-import/import.rb:458:in `import'
	from load_person_data.rb:194:in `block in <main>'
```

**Comment by MattMencel** [comment]
Yep, it's because my test/dev DB is SQLite3.  I ran a test import on my prod DB with 1 value entry and it worked fine.

Just curious what the best practice might be in this situation?  Don't use SQLite3 as my test/dev db?  :)

**Comment by MattMencel** [comment]
I think I answered my own question here.... I'll just avoid using the on_duplicate_key_update on my test DB.

**Comment by jkowens** [comment]
You got it 🙂

Unfortunately SQLite3 doesn’t have support for on duplicate key update.

**Comment by khiav223577** [comment]
I encountered this exception too.
It seems that `logger` may be nil in some cases.
https://github.com/zdennis/activerecord-import/blob/bab91e27066d2126420d41f6c7c45167ba389880/lib/activerecord-import/adapters/abstract_adapter.rb#L50

Could it use `puts` to print warnings if `logger` doesn't exist?
It throws `NoMethodError` now, which make it more difficult to find out what really going on.

**Comment by jkowens** [comment]
@khiav223577 thanks for the extra information! For now I think I'll just check if logger is initialized before trying to warn.

**Comment by jkowens** [comment]
This fix has just been released in v1.0.2 🚀
